### PR TITLE
Added HTTP response code 418, according to https://tools.ietf.org/html/rfc2324#section-2.3.2

### PR DESCRIPTION
**Pull request by andy-b-84**
# Pull Request Checklist
- [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
- [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
- [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
- [ ] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
- [x] Have you added copyright headers to new files?
- [x] Have you checked that both Scala and Java APIs are updated?
- [x] Have you updated the documentation for both Scala and Java sections?
- [x] Have you added tests for any changed functionality?
# Helpful things
## Fixes

Fixes no bug, this is a new feature.
## Purpose

Implements https://tools.ietf.org/html/rfc2324#section-2.3.2 : HTTP status 418 "I'm a teapot".
## Background Context

I think having a full list of available HTTP status codes is wonderful for developers (myself included).
I was sad to see this code missing.
## References

https://tools.ietf.org/html/rfc2324#section-2.3.2 

…l/rfc2324#section-2.3.2

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @andy-b-84,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: framework/src/play/src/main/scala/play/api/http/StandardValues.scala [hunk 1]
```diff
@@ -195,6 +195,7 @@ trait Status {
   val UNSUPPORTED_MEDIA_TYPE = 415
   val REQUESTED_RANGE_NOT_SATISFIABLE = 416
   val EXPECTATION_FAILED = 417
+  val I_M_A_TEAPOT = 418
```
Comment (gmethvin) [inline framework/src/play/src/main/scala/play/api/http/StandardValues.scala]: `IM_A_TEAPOT`?

File: framework/src/play/src/main/scala/play/api/mvc/Results.scala [hunk 2]
```diff
@@ -574,6 +574,9 @@ trait Results {
   /** Generates a ‘417 EXPECTATION_FAILED’ result. */
   val ExpectationFailed = new Status(EXPECTATION_FAILED)
 
+  /** Generates a ‘418 I_M_A_TEAPOT’ result. */
+  val IMATeapot = new Status(I_M_A_TEAPOT)
```
Comment (gmethvin) [inline framework/src/play/src/main/scala/play/api/mvc/Results.scala]: `ImATeapot`

**Comment by gmethvin** [comment]
Hi @andy-b-84, it would be nice to have a test for this.

We should test with an entity-body that is short and stout and with one that is not.

File: framework/src/play/src/main/scala/play/api/mvc/Results.scala (hunk 2 above)
Comment (andy-b-84) [inline framework/src/play/src/main/scala/play/api/mvc/Results.scala]: Received, thanks for your review :) (regarding the 3 points)

**Comment by typesafehub-validator** [comment]
Hi @andy-b-84,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by andy-b-84** [comment]
I didn't squash my commits before asking this PR (of course, I read the guidelines AFTER contributing, shame on me :) )
Do you want to refuse that PR, for me to force push on my master branch, then make a new PR?

**Comment by gmethvin** [comment]
@andy-b-84 We can squash the commits through github now, so you don't have to squash them yourself. I'd rather not create another PR; we generally prefer you force push to your branch to update if you need to.

**Comment by andy-b-84** [comment]
Great :)
I've tried to sign the CLA yesterday, nothing happened when I clicked the "submit" button on Lightbend's website. I just thought "OK, maybe it's some kind of Cron task or whatever".
I tried right now : same result.
I guess the form on https://www.lightbend.com/contribute/cla/verify doesn't work : it POSTs something to this URL, which answers with a 200 HTTP status code, but apparently something is broken between this website and Github.

**Comment by andy-b-84** [comment]
OK, I have signed the Lightbend CLA, finally!
I did it by connecting with a private navigation window ^^"

I guess it needs time fot Lightbend's system to tell Github?

**Comment by wsargent** [comment]
@andy-b-84 sorry about this, there was a problem last week -- can you try again and it should clear now?

**Comment by andy-b-84** [comment]
OK, I just signed the CLA (for the 3rd time now ^^" not that it hurts, of course).

Anything on your end?

**Comment by andy-b-84** [comment]
Hi,

I signed the CLA 2 weeks ago, but there still seem to be a problem. Can I do anything to help?

**Comment by andy-b-84** [comment]
Hi, still nothing?

**Comment by andy-b-84** [comment]
Hmm...
I just signed in using a private nav window on Chrome, I was able to sign the CLA, I even received the mail from mailer@lightbend.com .

**Comment by benmccann** [comment]
Yes, I can see that you signed here: http://www.lightbend.com/contribute/cla/check/andy-b-84

**Comment by benmccann** [comment]
From wikipedia:
>This code was defined in 1998 as one of the traditional IETF April Fools' jokes, in RFC 2324, Hyper Text Coffee Pot Control Protocol, and is not expected to be implemented by actual HTTP servers. The RFC specifies this code should be returned by teapots requested to brew coffee.[49] This HTTP status is used as an Easter egg in some websites, including Google.com.[50]

**Pull request merged**